### PR TITLE
Implement most ini_*() functions

### PR DESCRIPTION
**Pull request by hikari-no-yume**

### CODE REVIEW

File: runner/src/world/ini.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                        file.write_all(b"=").unwrap();
+                        file.write_all(new_value).unwrap();
+                        file.write_all(b"\r\n").unwrap();
+                        file.write_all(&buf).unwrap();
+                        return None;
+                    }
+                }
+            }
+
+            if found_section && &buf[0..key.len()] == key && buf.get(key.len()) == Some(&b'=') {
```
Comment (rpjohnst) [inline runner/src/world/ini.rs]: It looks like this panics when the current line is shorter than the key we're looking for? E.g. searching for `loooooong` in this file:

```rust
[section]
short=0
loooooong=1
```

Comment (hikari-no-yume) [reply runner/src/world/ini.rs]: Oh, somehow I thought ranges on slices were kind about this… maybe it's just ones where it's partially unbounded.

That explains why `Vec::get` exists for ranges I guess.

Comment (hikari-no-yume) [reply runner/src/world/ini.rs]: I fixed this now, and also made the section and key matching a bit prettier.

**Comment by hikari-no-yume** [comment]
I got a little carried away and made [some improvements](https://github.com/hikari-no-yume/dejavu/compare/8160ce2..dd41e85b0dcb315c5c9080259bbb59c61b36bcd3)…

**Comment by rpjohnst** [comment]
Looks good!

**Pull request merged**